clang/lib/CIR/Dialect/IR/CIRDialect.cpp

```diff
@@ -1534,14 +1550,21 @@ void cir::FuncOp::print(OpAsmPrinter &p) {
       p, *this,
       {getSymVisibilityAttrName(), getAliaseeAttrName(),
        getFunctionTypeAttrName(), getLinkageAttrName(), getBuiltinAttrName(),
-       getNoProtoAttrName()});
+       getNoProtoAttrName(), getExtraAttrsAttrName()});
```

Comment:
I managed to skip printing the extraAttr in a general way and do a custom printing below for CIR. Unfortunately, similar change will be needed to MLIR printing for the same behavior for LLVM dialect printing. For now I'm skipping that and the LLVM dialect will look like below with an empty extra attribute.

```
module {
  llvm.func @foo() attributes {cir.extra_attrs = #cir<>} {
    %0 = llvm.mlir.constant(true) : i8
    %1 = llvm.mlir.constant(1 : index) : i64
    %2 = llvm.alloca %1 x i8 {alignment = 1 : i64} : (i64) -> !llvm.ptr<i8>
    llvm.store %0, %2 : !llvm.ptr<i8>
    llvm.return
  }
}
```

Reply:
Why do we want cir.extra_attrs to survive into LLVM dialect? I guess I missing the context for this specific behavior.

Reply:
Sorry for the confusion. We don't want the slip, but we don't want to change MLIR code to specialcase for CIR attributes like above, right? The MLIR code is https://github.com/llvm/clangir/blob/c1fa60b4cde512964544ab66404dea79dbc5dcb4/mlir/lib/Dialect/LLVMIR/IR/LLVMDialect.cpp#L2150

Reply:
What I mean is, why CIR attributes are showing up in a llvm func? Shouldn't they be gone by the point we are printing a llvm func? I would expect to see a llvm attribute at this point. It's fine if this is just transitional, not expecting you to change more things as part of this patch, just trying to understand.

Reply:
The extra function attribute is plumbed all the way through LLVM dialect to LLVM IR. They stay in the original form unchanged on LLVM dialect and they get transformed to LLVM attributes by the time LLVM IR is generated.

I guess the LLVM dialect printing isn't super important at this time, and perhaps there isn't a strong demand to serialize/deseralize them?

Reply:
Thanks for the explanation, I now understand how this works! For the LLVM dialect printing: agreed.